### ISSUE TITLE
Deeply Nested Placeholder keys are not replaced

### ISSUE THREAD

**Issue by xpaulnim** [aws/aws-step-functions-data-science-sdk-python#92]
For instance, in the following
```json
{
  "InputDataConfig": [
    {
      "ChannelName": "training",
      "DataSource": {
        "S3DataSource": {
          "S3DataDistributionType": "FullyReplicated",
          "S3DataType": "S3Prefix",
          "S3Uri": "$.Payload.cloned_config"
        }
      }
    }
  ]
}
```
where S3Uri is either `ExecutionInput` or `StepInput` should be replaced with `S3Uri.$`

**Comment by wong-a** [comment]
Should be addressed by https://github.com/aws/aws-step-functions-data-science-sdk-python/pull/142

**Comment by shivlaks** [comment]
#142 has been merged and will be included in the next released version (`v2.2.0`)
